extensions/wikia/Email/SpecialSendEmailController.class.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+namespace Email;
+
+
+class SpecialSendEmailController extends \WikiaSpecialPageController {
+
+	const DEFAULT_TEMPLATE_ENGINE = \WikiaResponse::TEMPLATE_ENGINE_MUSTACHE;
+	const REQUIRED_PERMISSION = "staff";
+
+	public function __construct() {
+		parent::__construct( 'SendEmail', '', $listed = false );
+	}
+
+	/**
+	 * Run before index. First make sure the user can access page, then set the title and
+	 * add the css.
+	 */
+	public function init() {
+		$this->assertCanAccess();
+		$this->setTitle();
+		$this->addCss();
+	}
+
+	/**
+	 * Makes sure the current user is staff and can therefore access this page. If not, display
+	 * a permission error page.
+	 * @throws \PermissionsError
+	 */
+	private function assertCanAccess() {
+		if ( !$this->wg->User->isStaff() ) {
+			throw new \PermissionsError( self::REQUIRED_PERMISSION );
+		}
+	}
+
+	/**
+	 * Set's the <h1> and <title> tags, and suppresses the generic "Special Page" subtitle.
+	 */
+	private function setTitle() {
+		$this->getContext()->getOutput()->setHTMLTitle( "Special Send Email" );
+		$this->getContext()->getOutput()->setPageTitle( "Special Send Email" );
+		$this->wg->SupressPageSubtitle = true;
+	}
+
+	/**
+	 * Adds stylesheet to output
+	 */
+	private function addCss() {
+		$this->response->addAsset( 'special_email_admin_css' );
+	}
+
+	/**
+	 * @template specialSendEmail
+	 */
+	public function index() {
+
+		if ( $this->wg->request->wasPosted() ) {
```

Comment:
This should also [protect against CSRF](https://one.wikia-inc.com/wiki/Security/Cross-site_request_forgery#How_to_implement_CSRF_protection_on_Wikia.27s_version_of_MediaWiki) using `User::matchEditToken` and having the form contain the user's edit token in a hidden field.